cps/kni.c

```diff
@@ -381,72 +381,121 @@ attr_get(struct cps_config *cps_conf,
 			RTE_LOG(INFO, GATEKEEPER, "cps update: multipath\n");
 	}
 	if (tb[RTA_DST]) {
-		if (family == AF_INET6) {
+		if (family == AF_INET) {
+			struct in_addr *addr =
+				mnl_attr_get_payload(tb[RTA_DST]);
+			rte_memcpy(&update->ip.v4,
+				addr, sizeof(update->ip.v4));
+			if (cps_conf->debug)
+				inet_print("dst", addr);
+		} else if (likely(family == AF_INET6)) {
 			struct in6_addr *addr =
 				mnl_attr_get_payload(tb[RTA_DST]);
+			rte_memcpy(&update->ip.v6,
+				addr, sizeof(update->ip.v6));
 			if (cps_conf->debug)
 				inet6_print("dst", addr);
 		} else {
-			struct in_addr *addr =
-				mnl_attr_get_payload(tb[RTA_DST]);
-			if (cps_conf->debug)
-				inet_print("dst", addr);
+			RTE_LOG(WARNING, GATEKEEPER,
+				"cps update: unknown address family %d at %s!\n",
+				family, __func__);
+			return;
 		}
+
+		dst_present = true;
 	}
 	if (tb[RTA_SRC]) {
-		if (family == AF_INET6) {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"cps update: the rtnetlink command has information (RTA_SRC) that we don't need or don't honor!\n");
```

Comment:
Moving this log message after the `if` statement below enables to add the source address to the message, and drop the need for checking `cps_conf->debug` here.

Analogous change applies to `RTA_PREFSRC`.

Reply:
Please incorporate the address into the log message in both cases.

Reply:
``inet_print()`` and ``inet6_print()`` will log the message for address information just before this log message, do we need to add this information in this log message as well?